### PR TITLE
feat: user register

### PR DESCRIPTION
**Pull request by samueldurantes**

### CODE REVIEW

File: src/modules/user/controllers/userController.ts [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
   throw new Error('Unimplemented')
 }
+
+export const register = async (request: Request, response: Response) => {
+  const { username, email, password }: User = request.body
+
+  const user = await _register({ username, email, password })
+
+  if (isLeft(user)) {
+    console.log(user.left)
```
Comment (algebraic-dev) [inline src/modules/user/controllers/userController.ts]: Remove console.log

File: src/server.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import userRoutes from '@user/router'
 export default () => {
   const app = express()
 
+  app.use(express.json())
   app.use(userRoutes)
```
Comment (algebraic-dev) [inline src/server.ts]: Add /auth in front of user routes.

File: test/user/register.test.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      .post('/register')
+      .send({
+        username: 'john',
+        email: 'johndoe@domain.com',
+        password: 'johndoe123'
+      })
+      .set('Accept', 'application/json')
+      .expect('Content-Type', /json/)
+      .expect(201)
+      .then((response) => done(response))
```
Comment (algebraic-dev) [inline test/user/register.test.ts]: You can apply eta reduction on .then and .catch. `a => b(a)` is equivalent to `b`

**Review by algebraic-dev — CHANGES_REQUESTED** [review]

**Comment by samueldurantes** [comment]
Done!

**Review by algebraic-dev — APPROVED** [review]

**Pull request merged**